global_body_planner/include/global_body_planner/rrt.h

```diff
@@ -0,0 +1,50 @@
+#ifndef RRT_H
+#define RRT_H
+
+#include "global_body_planner/planner_class.h"
+#include <chrono>
+
+#define TRAPPED 0
+#define ADVANCED 1
+#define REACHED 2
+
+class RRTClass
+{
+    public:
+        //constructor
+        RRTClass();
+
+        //destructor
+        ~RRTClass();
+
+        virtual int extend(PlannerClass &T, State s, Ground& ground, int direction);
+        std::vector<int> pathFromStart(PlannerClass &T, int s);
+        void printPath(PlannerClass &T, std::vector<int> path);
+        void buildRRT(Ground &ground, State s_start, State s_goal, std::vector<State> &state_sequence, std::vector<Action> &action_sequence);
+        void getStatistics(double &plan_time, int &success_var, int &vertices_generated, double &time_to_first_solve, std::vector<double> &cost_vector, std::vector<double> &cost_vector_times, double& path_duration);
+
+        bool newConfig(State s, State s_near, State &s_new, Action &a_new, Ground &ground, int direction);
+
+        std::vector<State> getStateSequence(PlannerClass &T, std::vector<int> path);
+        std::vector<Action> getActionSequence(PlannerClass &T, std::vector<int> path);
```

Comment:
Why the need to accept a PlannerClass obj as an argument here? Could this be solved by subclassing PlannerClass?